sql/src/test/java/io/crate/execution/jobs/JobExecutionContextTest.java

```diff
@@ -151,4 +152,27 @@ public void testFailureClosesAllSubContexts() throws Exception {
 
         assertThat(size, is(0));
     }
+
+    @Test
+    public void testEnablingProfilingGathersExecutionTimes() throws Throwable {
+        JobExecutionContext.Builder builder =
+            new JobExecutionContext.Builder(UUID.randomUUID(), coordinatorNode, Collections.emptySet(), mock(JobsLogs.class));
+        builder.enableProfiling(true);
+
+        AbstractExecutionSubContextTest.TestingExecutionSubContext ctx1 = new AbstractExecutionSubContextTest.TestingExecutionSubContext(1);
+        builder.addSubContext(ctx1);
+        AbstractExecutionSubContextTest.TestingExecutionSubContext ctx2 = new AbstractExecutionSubContextTest.TestingExecutionSubContext(2);
+        builder.addSubContext(ctx2);
+        JobExecutionContext jobExecutionContext = builder.build();
+
+        jobExecutionContext.start();
+        // fake execution time so we can sure the measurement is > 0
+        Thread.sleep(5L);
```

Comment:
We could use a custom `TimeMeasurable` here which advances the time on start. Or maybe just decrease the sleep time. It's quite long.

Reply:
Hm, 5ms is not *that* long.
Being able to mock the TimeMeasurable that is used in the ProfilingContext in the JobExecutionContext would require quite some changes. Not sure if it would be worth the effort, tbh.

Reply:
Sorry, for a second (pun intended) I forgot that Thread.sleep uses milliseconds.